lib/utils/config.ts

```diff
@@ -165,3 +176,48 @@ export async function validateDevOptionAndWarn(
 		and exposes network ports such as 2375 that allows unencrypted access to balenaEngine.
 		Therefore, development mode should only be used in private, trusted local networks.`);
 }
+
+/**
+ * Chech whether the `--secureBoot` option of commands related to OS configuration
+ * such as `os configure` and `config generate` is compatible with a given
+ * device type and balenaOS version, and print a warning regarding the
+ * consequences of using that option.
+ */
+export async function validateSecureBootOptionAndWarn(
+	secureBoot?: boolean,
+	deviceManifest?: any,
+	version?: string,
+	logger?: import('./logger'),
+) {
+	if (!secureBoot) {
+		return;
+	}
+
+	const deviceManifestObj: { [index: string]: any } = deviceManifest;
+	const deviceType = deviceManifestObj['yocto']['machine'];
+
+	const { ExpectedError } = await import('../errors');
+	const supportedDeviceTypes = ['generic-amd64'];
+	if (!supportedDeviceTypes.includes(`${deviceType}`)) {
+		throw new ExpectedError(
+			`Error: The '--secureBoot' option is not supported for the '${deviceType}' device Type. Supported device types include ${supportedDeviceTypes.join(
+				', ',
+			)}.`,
+		);
+	}
+
+	const supportedFromVersion = '2.114.21';
```

Comment:
> Ideally this should come from the DT contract (or the device-type.json :disappointed: ), so that we don't need to release new versions of the CLI every time we add support for secure boot to an extra DT.
> 
> For now, let's just move this const definition outside of the function at the top level to raise its importance vs other vars.

Reply:
I used this in the update, thanks.